registry/endpoint.go

```diff
@@ -54,12 +54,12 @@ func NewEndpoint(index *IndexInfo) (*Endpoint, error) {
 	return endpoint, nil
 }
 
-func validateEndpoint(endpoint *Endpoint) error {
+func validateEndpoint(endpoint *Endpoint) (err error) {
 	log.Debugf("pinging registry endpoint %s", endpoint)
 
 	// Try HTTPS ping to registry
 	endpoint.URL.Scheme = "https"
-	if _, err := endpoint.Ping(); err != nil {
+	if endpoint.RegInfo, err = endpoint.Ping(); err != nil {
```

Comment:
Should be okay since a ping attempt is made again after this, and if that one fails then the endpoint is no good anyway.